central/sensor/service/pipeline/nodeinventory/two_pipelines_test.go

```diff
@@ -0,0 +1,404 @@
+				},
+			},
+			setUpMocks: func(t *testing.T, m *usedMocks) {
+				gomock.InOrder(
+					// node inventory arrives
+					m.nodeDatastore.EXPECT().GetNode(gomock.Any(), gomock.Eq(nodeID)).Return(nil, false, nil),
+
+					// node arrives
+					m.clusterStore.EXPECT().GetClusterName(gomock.Any(), gomock.Eq(clusterID)).Return(clusterID, true, nil),
+					m.cveDatastore.EXPECT().EnrichNodeWithSuppressedCVEs(gomock.Any()),
+					m.riskStorage.EXPECT().UpsertRisk(gomock.Any(), gomock.Any()).Times(2).Return(nil),
+					m.nodeDatastore.EXPECT().UpsertNode(gomock.Any(), gomock.Any()).Return(nil),
+
+					// check what got stored in the DB
+					m.nodeDatastore.EXPECT().GetNode(gomock.Any(), gomock.Eq(nodeID)).AnyTimes().Return(nodeWithScanWithKernelV1, true, nil),
+				)
+			},
+			wantNodeExists:            true,
+			wantKernelVersionNode:     "v1",
+			wantKernelVersionNodeScan: "v1",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			tt.mocks = usedMocks{
+				clusterStore:      clusterDatastoreMocks.NewMockDataStore(ctrl),
+				nodeDatastore:     nodeDatastoreMocks.NewMockDataStore(ctrl),
+				cveDatastore:      nodeCVEDataStoreMocks.NewMockDataStore(ctrl),
+				deploymentStorage: mocks.NewMockDataStore(ctrl),
+				imageStorage:      imageStoreMock.NewMockDataStore(ctrl),
+				riskStorage:       riskStoreMock.NewMockDataStore(ctrl),
+				updater:           updaterMocks.NewMockUpdater(ctrl),
+			}
+			tt.riskManager = manager.New(
+				tt.mocks.nodeDatastore,
+				tt.mocks.deploymentStorage,
+				tt.mocks.imageStorage,
+				tt.mocks.riskStorage,
+				&mockNodeScorer{},
+				&mockComponentScorer{},
+				&mockDeploymentScorer{},
+				&mockImageScorer{},
+				&mockComponentScorer{},
+
+				ranking.ClusterRanker(),
+				ranking.NamespaceRanker(),
+				ranking.ComponentRanker(),
+				ranking.NodeComponentRanker(),
+
+				tt.mocks.updater,
+			)
+			creator := func() (string, scanners.NodeScannerCreator) {
+				return "fake", func(integration *storage.NodeIntegration) (types.NodeScanner, error) {
+					return &fakeNodeScanner{}, nil
+				}
+			}
+			tt.enricher = nodeEnricher.NewWithCreator(tt.mocks.cveDatastore, metrics.CentralSubsystem, creator)
+			err := tt.enricher.UpsertNodeIntegration(&storage.NodeIntegration{
+				Id:   "1",
+				Name: "dummy-scanner",
+				Type: "fake",
+				IntegrationConfig: &storage.NodeIntegration_Clairify{Clairify: &storage.ClairifyConfig{
+					Endpoint:           "abc",
+					GrpcEndpoint:       "",
+					NumConcurrentScans: 0,
+				}},
+			})
+			assert.NoError(t, err)
+
+			if tt.setUpMocks != nil {
+				tt.setUpMocks(t, &tt.mocks)
+			}
+			pNode := nodes.NewPipeline(tt.mocks.clusterStore, tt.mocks.nodeDatastore, tt.enricher, tt.riskManager)
+			pNodeInv := newPipeline(tt.mocks.clusterStore, tt.mocks.nodeDatastore, tt.enricher, tt.riskManager)
+
+			var lastErr error
+			for i, op := range tt.operations {
```

Comment:
This block looks a little odd to me. 
1. `lastErr` will only have the error for the last step
2. There is no way to check intermediate step failures, since their errors will be overwritten.

Wouldn't it be valid to check also something like the following ordered operations:
1. NodeV2 with error
2. NoveV1 without error